version: "3.8"

services:
  postgres:
    image: postgres:12-alpine
    restart: on-failure
    ports:
      - "5432:5432"
    networks:
      - common-shared-network
    environment:
      POSTGRES_DB: overflow
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: 12345678

  lavalink:
    image: fredboat/lavalink:dev
    restart: on-failure
    volumes:
      - ./lavalink.yml:/opt/Lavalink/application.yml
      - ./lavalink_logs:/opt/Lavalink/logs
    ports:
      - "2333:2333"
    networks:
      - common-shared-network

  bot:
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - ./logs:/bot/logs
      - .:/bot:ro
    tty: true
    depends_on:
      - lavalink
      - postgres
    environment:
      LAVALINK_HOST: lavalink
      LAVALINK_PORT: 2333
      LAVALINK_PASSWORD: youshallnotpass
    env_file:
      - .env
    links:
      - postgres
      - lavalink
    restart: on-failure
    networks:
      - common-shared-network

networks:
  common-shared-network:
